<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Hyunwoo Kwon" />

<meta name="date" content="2025-03-11" />

<title>Assignment 2: Gene Expression Analysis and Preliminary ORA</title>

<script src="BCB420A2_files/header-attrs-2.29/header-attrs.js"></script>
<script src="BCB420A2_files/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="BCB420A2_files/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="BCB420A2_files/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="BCB420A2_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="BCB420A2_files/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="BCB420A2_files/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="BCB420A2_files/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="BCB420A2_files/tocify-1.9.1/jquery.tocify.js"></script>
<script src="BCB420A2_files/navigation-1.1/tabsets.js"></script>
<script src="BCB420A2_files/navigation-1.1/codefolding.js"></script>
<link href="BCB420A2_files/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="BCB420A2_files/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

.sourceCode .row {
  width: 100%;
}
.sourceCode {
  overflow-x: auto;
}
.code-folding-btn {
  margin-right: -30px;
}
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Assignment 2: Gene Expression Analysis and
Preliminary ORA</h1>
<h4 class="author">Hyunwoo Kwon</h4>
<h4 class="date">2025-03-11</h4>

</div>


<div id="set-up" class="section level1" number="1">
<h1><span class="header-section-number">1</span> <strong>Set
Up</strong></h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># Install dependencies</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">error =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: package &#39;BiocManager&#39; was built under R version 4.3.3</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;edgeR&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>    BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;edgeR&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: package(s) not installed when version(s) same as or greater than current; use
##   `force = TRUE` to re-install: &#39;edgeR&#39;</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;limma&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>    BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;limma&quot;</span>)</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;gprofiler2&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>    BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;gprofiler2&quot;</span>)</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;ggplot2&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;pheatmap&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;pheatmap&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: package &#39;pheatmap&#39; was built under R version 4.3.3</code></pre>
<p>Loading Normalized Data from A1</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co"># Load normalized count matrix from Assignment 1</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>destfile <span class="ot">&lt;-</span> <span class="st">&quot;normalized_counts.rds&quot;</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="fu">saveRDS</span>(normalized_counts, <span class="st">&quot;normalized_counts.rds&quot;</span>)</span></code></pre></div>
<pre><code>## Error: object &#39;normalized_counts&#39; not found</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(destfile)) {</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>  <span class="fu">download.file</span>(</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>    <span class="at">url =</span> <span class="st">&quot;https://raw.githubusercontent.com/bcb420-2025/Hyunwoo_Kwon/main/A1/normalized_counts.rds&quot;</span>, </span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>    <span class="at">destfile =</span> destfile</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>  )</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>}</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a><span class="co"># Load data</span></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a>normalized_counts <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(destfile)</span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a><span class="fu">head</span>(normalized_counts)</span></code></pre></div>
<pre><code>##         Patient_01_NT Patient_01_T Patient_02_T Patient_03_NT Patient_03_T
## SAMD11      -2.052095     2.552639    2.3444865     -0.625362   2.82391841
## NOC2L        6.137338     4.826989    5.6663978      6.142678   5.33916090
## KLHL17       2.406962     1.043656    0.8002017      1.074977   1.18348912
## PLEKHN1      5.185919     1.513128    0.6746758      4.913654  -1.31881663
## HES4         2.471091     2.098080    2.3444865      1.374528   2.58785343
## ISG15        1.117481     2.735502    3.2595888      0.374565  -0.09655923
##         Patient_05_T Patient_06_NT Patient_06_T Patient_07_T Patient_08_NT
## SAMD11     1.4277312  -0.005316542     2.924497     3.531057     1.7672581
## NOC2L      5.7172045   6.363823355     5.491855     4.926190     5.4396578
## KLHL17     2.1157739   0.869109367     1.898038     1.041710     0.7672859
## PLEKHN1    3.7813399   5.137549040     2.602572    -2.073415    -2.2323247
## HES4       0.9582595   0.994635928     1.787009     1.926212    -0.6476589
## ISG15      0.4277664   0.217062300     1.465088     1.589183     1.9371800
##         Patient_08_T Patient_09_T Patient_10_NT Patient_10_T Patient_11_NT
## SAMD11     2.1084884    1.5817884    0.97705677    2.8881980     1.8130687
## NOC2L      5.2736715    5.2006692    6.11451355    6.1485268     5.2724758
## KLHL17     0.1910115    0.8899302    2.02433765    1.2815666     0.9551096
## PLEKHN1    0.6060283    0.1895224    4.57791667    2.8881980    -0.2374727
## HES4       2.3064249    1.7744295    0.52961529    1.8221228     0.6555606
## ISG15      1.9279191    4.5469954    0.07020827    0.6295121     1.4145279
##         Patient_11_T Patient_12_NT Patient_12_T Patient_13_NT Patient_13_T
## SAMD11      1.369369   -0.03899091    3.8408359     0.5076399     1.538661
## NOC2L       5.769556    6.22300790    5.2331492     6.0018918     6.282791
## KLHL17      1.990844    1.48450759    0.5607856     1.1291049     2.304183
## PLEKHN1     2.423797    4.16058942   -1.2464089     4.5620250     3.024067
## HES4        2.061232    2.00532949    1.7075917     1.5076065     1.801690
## ISG15       4.575787    2.60478354    2.1457052     0.7140818     2.386644
##         Patient_14_NT Patient_14_T Patient_15_NT Patient_15_T Patient_16_NT
## SAMD11     2.14465078     2.005895     -1.332445    1.5607665    -0.5309677
## NOC2L      5.36879969     6.093335      6.161840    5.2611766     6.4537896
## KLHL17     1.19358037     1.099025      1.082399    1.6982671     1.2059021
## PLEKHN1    0.09409202     3.228274      4.775845    1.2388465     5.1693378
## HES4       0.98713586     2.742851      1.082399    0.6982963     0.4689639
## ISG15      2.53459222     3.590841      1.404318    1.9937176     0.3434392
##         Patient_16_T Patient_18_NT Patient_18_T Patient_19_NT Patient_19_T
## SAMD11     1.2151425     0.8673356    2.4102375    -1.1778777     1.925052
## NOC2L      5.4513021     6.0711575    5.6732559     6.2634701     6.359085
## KLHL17     0.5147283     1.5835223    1.4419634     0.6293241     1.127563
## PLEKHN1   -0.4852054     4.4086614    4.1081966     4.9614624     2.243017
## HES4       0.5147283     1.7603965    0.8253106     0.3244839     2.020628
## ISG15      1.0996687     0.2299346    1.7763752     1.3244461     5.377256
##         Patient_20_NT Patient_20_T Patient_21_NT Patient_21_T Patient_22_T
## SAMD11      0.3028933     1.668536    0.03931733   -0.9093418     1.526401
## NOC2L       5.7778467     6.441646    6.37142274    6.5381550     5.710794
## KLHL17      1.6121755     2.268438    1.81685963    1.7425853     1.569468
## PLEKHN1     4.8154315     2.340950    5.03922805    0.3706615     0.289406
## HES4        1.4797282     2.396444    1.47983172    3.4124173     1.388900
## ISG15       0.9887556     2.409991    1.40183109    3.5499200     1.729930
##         Patient_23_NT Patient_23_T Patient_24_NT Patient_24_T Patient_22_NT
## SAMD11     -1.8602587    1.8911061    -0.5857702    0.5369419      4.226486
## NOC2L       6.7895524    5.6275915     6.2532926    6.2149498      4.226486
## KLHL17      1.1838332    1.2020509     1.4141231    1.7068306      4.226486
## PLEKHN1     5.3267518    0.2878172     4.7891305    2.2593624      4.226486
## HES4        1.7832811    1.1723044     1.0464017    1.3024498      4.226486
## ISG15       0.7833086    1.6534192     0.4845446    4.2428595      4.226486</code></pre>
<p>Creating a design Matrix</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="co"># Get the correct number of samples</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>num_samples <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">colnames</span>(normalized_counts)[<span class="sc">-</span><span class="dv">1</span>]) </span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a><span class="co"># Assuming 2 conditions per patient, adjust dynamically</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a>num_patients <span class="ot">&lt;-</span> <span class="fu">floor</span>(num_samples <span class="sc">/</span> <span class="dv">2</span>) </span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a><span class="co"># Automatically assign conditions</span></span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a>conditions <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&quot;Tumor&quot;</span>, <span class="st">&quot;Non-Tumor&quot;</span>), <span class="at">length.out =</span> num_samples) </span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a><span class="co"># Generate patient IDs dynamically</span></span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a>patients <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="fu">paste0</span>(<span class="st">&quot;P&quot;</span>, <span class="dv">1</span><span class="sc">:</span>num_patients), <span class="at">each =</span> <span class="dv">2</span>, <span class="at">length.out =</span> num_samples)</span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a></span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a><span class="co"># Verify that both vectors have the correct length</span></span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">length</span>(conditions))</span></code></pre></div>
<pre><code>## [1] 39</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">length</span>(patients))</span></code></pre></div>
<pre><code>## [1] 39</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="co"># Create a design data frame</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>  <span class="at">sample =</span> <span class="fu">colnames</span>(normalized_counts)[<span class="sc">-</span><span class="dv">1</span>],  <span class="co"># Column names as sample names</span></span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a>  <span class="at">patient =</span> patients,</span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a>  <span class="at">condition =</span> conditions</span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a>)</span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a></span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a><span class="co"># Convert to model matrix</span></span>
<span id="cb15-9"><a href="#cb15-9" tabindex="-1"></a>model_matrix <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> patient <span class="sc">+</span> condition, <span class="at">data =</span> design)</span>
<span id="cb15-10"><a href="#cb15-10" tabindex="-1"></a></span>
<span id="cb15-11"><a href="#cb15-11" tabindex="-1"></a><span class="co"># View model matrix</span></span>
<span id="cb15-12"><a href="#cb15-12" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">head</span>(model_matrix))</span></code></pre></div>
<pre><code>##   (Intercept) patientP10 patientP11 patientP12 patientP13 patientP14 patientP15
## 1           1          0          0          0          0          0          0
## 2           1          0          0          0          0          0          0
## 3           1          0          0          0          0          0          0
## 4           1          0          0          0          0          0          0
## 5           1          0          0          0          0          0          0
## 6           1          0          0          0          0          0          0
##   patientP16 patientP17 patientP18 patientP19 patientP2 patientP3 patientP4
## 1          0          0          0          0         0         0         0
## 2          0          0          0          0         0         0         0
## 3          0          0          0          0         1         0         0
## 4          0          0          0          0         1         0         0
## 5          0          0          0          0         0         1         0
## 6          0          0          0          0         0         1         0
##   patientP5 patientP6 patientP7 patientP8 patientP9 conditionTumor
## 1         0         0         0         0         0              1
## 2         0         0         0         0         0              0
## 3         0         0         0         0         0              1
## 4         0         0         0         0         0              0
## 5         0         0         0         0         0              1
## 6         0         0         0         0         0              0</code></pre>
<p>Differentiate expressing using Limma</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a><span class="co"># Ensure normalized_counts is a data frame</span></span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>normalized_counts <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(normalized_counts)</span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a><span class="co"># Check if hgnc_symbol exists as a column or rownames</span></span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a><span class="cf">if</span> (<span class="st">&quot;hgnc_symbol&quot;</span> <span class="sc">%in%</span> <span class="fu">colnames</span>(normalized_counts)) {</span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a>  gene_names <span class="ot">&lt;-</span> normalized_counts<span class="sc">$</span>hgnc_symbol</span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a>} <span class="cf">else</span> <span class="cf">if</span> (<span class="sc">!</span><span class="fu">is.null</span>(<span class="fu">rownames</span>(normalized_counts))) {</span>
<span id="cb17-8"><a href="#cb17-8" tabindex="-1"></a>  gene_names <span class="ot">&lt;-</span> <span class="fu">rownames</span>(normalized_counts)</span>
<span id="cb17-9"><a href="#cb17-9" tabindex="-1"></a>} <span class="cf">else</span> {</span>
<span id="cb17-10"><a href="#cb17-10" tabindex="-1"></a>  <span class="fu">stop</span>(<span class="st">&quot;Error: &#39;hgnc_symbol&#39; not found as a column or row names in normalized_counts&quot;</span>)</span>
<span id="cb17-11"><a href="#cb17-11" tabindex="-1"></a>}</span>
<span id="cb17-12"><a href="#cb17-12" tabindex="-1"></a></span>
<span id="cb17-13"><a href="#cb17-13" tabindex="-1"></a><span class="co"># Convert to matrix</span></span>
<span id="cb17-14"><a href="#cb17-14" tabindex="-1"></a>expression_data <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(normalized_counts[, <span class="sc">-</span><span class="dv">1</span>])</span>
<span id="cb17-15"><a href="#cb17-15" tabindex="-1"></a><span class="fu">rownames</span>(expression_data) <span class="ot">&lt;-</span> gene_names  <span class="co"># Assign gene names</span></span>
<span id="cb17-16"><a href="#cb17-16" tabindex="-1"></a></span>
<span id="cb17-17"><a href="#cb17-17" tabindex="-1"></a><span class="co"># Fit linear model</span></span>
<span id="cb17-18"><a href="#cb17-18" tabindex="-1"></a>fit <span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">lmFit</span>(expression_data, model_matrix)</span>
<span id="cb17-19"><a href="#cb17-19" tabindex="-1"></a></span>
<span id="cb17-20"><a href="#cb17-20" tabindex="-1"></a><span class="co"># Apply empirical Bayes moderation</span></span>
<span id="cb17-21"><a href="#cb17-21" tabindex="-1"></a>fit <span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">eBayes</span>(fit)</span>
<span id="cb17-22"><a href="#cb17-22" tabindex="-1"></a></span>
<span id="cb17-23"><a href="#cb17-23" tabindex="-1"></a><span class="co"># Extract differentially expressed genes</span></span>
<span id="cb17-24"><a href="#cb17-24" tabindex="-1"></a>limma_results <span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">topTable</span>(fit, <span class="at">coef =</span> <span class="st">&quot;conditionTumor&quot;</span>, <span class="at">adjust.method =</span> <span class="st">&quot;BH&quot;</span>, <span class="at">number=</span><span class="fu">nrow</span>(expression_data))</span>
<span id="cb17-25"><a href="#cb17-25" tabindex="-1"></a></span>
<span id="cb17-26"><a href="#cb17-26" tabindex="-1"></a><span class="co"># View summary</span></span>
<span id="cb17-27"><a href="#cb17-27" tabindex="-1"></a><span class="fu">sprintf</span>(<span class="st">&quot;Number of genes with p &lt; 0.05: %d&quot;</span>, <span class="fu">sum</span>(limma_results<span class="sc">$</span>P.Value <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span></code></pre></div>
<pre><code>## [1] &quot;Number of genes with p &lt; 0.05: 1228&quot;</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="fu">sprintf</span>(<span class="st">&quot;Number of genes that pass correction (FDR &lt; 0.05): %d&quot;</span>, <span class="fu">sum</span>(limma_results<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span></code></pre></div>
<pre><code>## [1] &quot;Number of genes that pass correction (FDR &lt; 0.05): 0&quot;</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="co"># Save results</span></span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a><span class="fu">write.csv</span>(limma_results, <span class="st">&quot;limma_DE_results.csv&quot;</span>)</span></code></pre></div>
<p>Differentiate expressiong using edgeR</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="co"># Define DGEList file from A1</span></span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a><span class="fu">remove.packages</span>(<span class="st">&quot;locfit&quot;</span>)</span></code></pre></div>
<pre><code>## Error in find.package(pkgs, lib): there is no package called &#39;locfit&#39;</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;locfit&quot;</span>, <span class="at">dependencies=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Error in contrib.url(repos, &quot;source&quot;): trying to use CRAN without setting a mirror</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>dge_file <span class="ot">&lt;-</span> <span class="st">&quot;post_norm_dge.rds&quot;</span></span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a><span class="co"># Check if file exists, otherwise download from GitHub</span></span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">file.exists</span>(dge_file)) {</span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a>  <span class="fu">download.file</span>(</span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a>    <span class="at">url =</span> <span class="st">&quot;https://raw.githubusercontent.com/bcb420-2025/Hyunwoo_Kwon/main/A1/post_norm_dge.rds&quot;</span>,</span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a>    <span class="at">destfile =</span> dge_file</span>
<span id="cb26-8"><a href="#cb26-8" tabindex="-1"></a>  )</span>
<span id="cb26-9"><a href="#cb26-9" tabindex="-1"></a>}</span>
<span id="cb26-10"><a href="#cb26-10" tabindex="-1"></a></span>
<span id="cb26-11"><a href="#cb26-11" tabindex="-1"></a><span class="co"># Load the DGEList object</span></span>
<span id="cb26-12"><a href="#cb26-12" tabindex="-1"></a>dge <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(dge_file)</span>
<span id="cb26-13"><a href="#cb26-13" tabindex="-1"></a></span>
<span id="cb26-14"><a href="#cb26-14" tabindex="-1"></a><span class="co"># Load edgeR library</span></span>
<span id="cb26-15"><a href="#cb26-15" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span></code></pre></div>
<pre><code>## Error: package or namespace load failed for &#39;edgeR&#39; in loadNamespace(j &lt;- i[[1L]], c(lib.loc, .libPaths()), versionCheck = vI[[j]]):
##  there is no package called &#39;locfit&#39;</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fu">estimateDisp</span>(dge, model_matrix)</span></code></pre></div>
<pre><code>## Error in estimateDisp(dge, model_matrix): could not find function &quot;estimateDisp&quot;</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a><span class="co"># Fit the model using quasi-likelihood estimation</span></span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">glmQLFit</span>(d, model_matrix)</span></code></pre></div>
<pre><code>## Error in glmQLFit(d, model_matrix): could not find function &quot;glmQLFit&quot;</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a><span class="co"># Perform Quasi-Likelihood F-Test for &quot;conditionTumor&quot;</span></span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a>qlf_test <span class="ot">&lt;-</span> <span class="fu">glmQLFTest</span>(fit, <span class="at">coef =</span> <span class="st">&quot;conditionTumor&quot;</span>)</span></code></pre></div>
<pre><code>## Error in glmQLFTest(fit, coef = &quot;conditionTumor&quot;): could not find function &quot;glmQLFTest&quot;</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a><span class="co"># Extract differentially expressed genes (DEGs)</span></span>
<span id="cb34-2"><a href="#cb34-2" tabindex="-1"></a>edgeR_results <span class="ot">&lt;-</span> <span class="fu">topTags</span>(qlf_test, <span class="at">n =</span> <span class="fu">nrow</span>(dge<span class="sc">$</span>counts))<span class="sc">$</span>table</span></code></pre></div>
<pre><code>## Error in topTags(qlf_test, n = nrow(dge$counts)): could not find function &quot;topTags&quot;</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a><span class="co"># Count significant genes</span></span>
<span id="cb36-2"><a href="#cb36-2" tabindex="-1"></a><span class="fu">sprintf</span>(<span class="st">&quot;Number of genes with p &lt; 0.05 (edgeR): %d&quot;</span>, <span class="fu">sum</span>(edgeR_results<span class="sc">$</span>PValue <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span></code></pre></div>
<pre><code>## Error: object &#39;edgeR_results&#39; not found</code></pre>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a><span class="fu">sprintf</span>(<span class="st">&quot;Number of genes passing correction (FDR &lt; 0.05): %d&quot;</span>, <span class="fu">sum</span>(edgeR_results<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>))</span></code></pre></div>
<pre><code>## Error: object &#39;edgeR_results&#39; not found</code></pre>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" tabindex="-1"></a><span class="co"># Save results to CSV</span></span>
<span id="cb40-2"><a href="#cb40-2" tabindex="-1"></a><span class="fu">write.csv</span>(edgeR_results, <span class="st">&quot;edgeR_DE_results.csv&quot;</span>)</span></code></pre></div>
<pre><code>## Error in eval(expr, p): object &#39;edgeR_results&#39; not found</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;ggplot2&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb42-2"><a href="#cb42-2" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb42-3"><a href="#cb42-3" tabindex="-1"></a>}</span>
<span id="cb42-4"><a href="#cb42-4" tabindex="-1"></a></span>
<span id="cb42-5"><a href="#cb42-5" tabindex="-1"></a><span class="co"># Load ggplot2</span></span>
<span id="cb42-6"><a href="#cb42-6" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb42-7"><a href="#cb42-7" tabindex="-1"></a></span>
<span id="cb42-8"><a href="#cb42-8" tabindex="-1"></a><span class="co"># Create volcano plot</span></span>
<span id="cb42-9"><a href="#cb42-9" tabindex="-1"></a><span class="fu">ggplot</span>(edgeR_results, <span class="fu">aes</span>(<span class="at">x =</span> logFC, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(PValue), <span class="at">color =</span> (FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>))) <span class="sc">+</span></span>
<span id="cb42-10"><a href="#cb42-10" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb42-11"><a href="#cb42-11" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;gray&quot;</span>, <span class="st">&quot;red&quot;</span>)) <span class="sc">+</span></span>
<span id="cb42-12"><a href="#cb42-12" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb42-13"><a href="#cb42-13" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;Volcano Plot&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Log Fold Change&quot;</span>, <span class="at">y =</span> <span class="st">&quot;-log10 P-Value&quot;</span>) <span class="sc">+</span></span>
<span id="cb42-14"><a href="#cb42-14" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>, <span class="at">color =</span> <span class="st">&quot;blue&quot;</span>)</span></code></pre></div>
<pre><code>## Error: object &#39;edgeR_results&#39; not found</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;pheatmap&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb44-2"><a href="#cb44-2" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;pheatmap&quot;</span>)</span>
<span id="cb44-3"><a href="#cb44-3" tabindex="-1"></a>}</span>
<span id="cb44-4"><a href="#cb44-4" tabindex="-1"></a><span class="co"># Load necessary library</span></span>
<span id="cb44-5"><a href="#cb44-5" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb44-6"><a href="#cb44-6" tabindex="-1"></a></span>
<span id="cb44-7"><a href="#cb44-7" tabindex="-1"></a><span class="co"># Select top 50 differentially expressed genes</span></span>
<span id="cb44-8"><a href="#cb44-8" tabindex="-1"></a>top50 <span class="ot">&lt;-</span> <span class="fu">rownames</span>(<span class="fu">head</span>(edgeR_results, <span class="dv">50</span>))</span></code></pre></div>
<pre><code>## Error: object &#39;edgeR_results&#39; not found</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" tabindex="-1"></a><span class="co"># Extract corresponding expression data</span></span>
<span id="cb46-2"><a href="#cb46-2" tabindex="-1"></a>heatmap_data <span class="ot">&lt;-</span> normalized_counts[top50, ]</span></code></pre></div>
<pre><code>## Error: object &#39;top50&#39; not found</code></pre>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" tabindex="-1"></a><span class="co"># Generate heatmap</span></span>
<span id="cb48-2"><a href="#cb48-2" tabindex="-1"></a><span class="fu">pheatmap</span>(heatmap_data, <span class="at">scale =</span> <span class="st">&quot;row&quot;</span>, </span>
<span id="cb48-3"><a href="#cb48-3" tabindex="-1"></a>         <span class="at">clustering_distance_rows =</span> <span class="st">&quot;euclidean&quot;</span>, </span>
<span id="cb48-4"><a href="#cb48-4" tabindex="-1"></a>         <span class="at">clustering_distance_cols =</span> <span class="st">&quot;euclidean&quot;</span>, </span>
<span id="cb48-5"><a href="#cb48-5" tabindex="-1"></a>         <span class="at">clustering_method =</span> <span class="st">&quot;complete&quot;</span>,</span>
<span id="cb48-6"><a href="#cb48-6" tabindex="-1"></a>         <span class="at">show_rownames =</span> <span class="cn">TRUE</span>, <span class="at">show_colnames =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Error: object &#39;heatmap_data&#39; not found</code></pre>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" tabindex="-1"></a><span class="co"># Install gprofiler2 if necessary</span></span>
<span id="cb50-2"><a href="#cb50-2" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;gprofiler2&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb50-3"><a href="#cb50-3" tabindex="-1"></a>    BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;gprofiler2&quot;</span>)</span>
<span id="cb50-4"><a href="#cb50-4" tabindex="-1"></a></span>
<span id="cb50-5"><a href="#cb50-5" tabindex="-1"></a><span class="co"># Load package</span></span>
<span id="cb50-6"><a href="#cb50-6" tabindex="-1"></a><span class="fu">library</span>(gprofiler2)</span>
<span id="cb50-7"><a href="#cb50-7" tabindex="-1"></a></span>
<span id="cb50-8"><a href="#cb50-8" tabindex="-1"></a><span class="co"># Extract significant genes</span></span>
<span id="cb50-9"><a href="#cb50-9" tabindex="-1"></a>significant_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(edgeR_results[edgeR_results<span class="sc">$</span>FDR <span class="sc">&lt;</span> <span class="fl">0.05</span>, ])</span></code></pre></div>
<pre><code>## Error: object &#39;edgeR_results&#39; not found</code></pre>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" tabindex="-1"></a><span class="co"># Perform enrichment analysis</span></span>
<span id="cb52-2"><a href="#cb52-2" tabindex="-1"></a>gost_results <span class="ot">&lt;-</span> <span class="fu">gost</span>(<span class="at">query =</span> significant_genes, <span class="at">organism =</span> <span class="st">&quot;hsapiens&quot;</span>, <span class="at">sources =</span> <span class="fu">c</span>(<span class="st">&quot;GO:BP&quot;</span>, <span class="st">&quot;KEGG&quot;</span>))</span></code></pre></div>
<pre><code>## Error: object &#39;significant_genes&#39; not found</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" tabindex="-1"></a><span class="co"># View results</span></span>
<span id="cb54-2"><a href="#cb54-2" tabindex="-1"></a>gost_results<span class="sc">$</span>result</span></code></pre></div>
<pre><code>## Error: object &#39;gost_results&#39; not found</code></pre>
<p><strong>Calculate p-values for each of the genes in your expression
set. How many genes were significantly differentially expressed? What
thresholds did you use and why? </strong>Multiple hypothesis testing -
correct your p-values using a multiple hypothesis correction method.
Which method did you use? And Why? How many genes passed correction?
<strong>Show the amount of differentially expressed genes using an MA
Plot or a Volcano plot. Highlight genes of interest. </strong>Visualize
your top hits using a heatmap. Do you conditions cluster together?
Explain why or why not. **Make sure all your figures have proper heading
and labels. Every figure included in the report should have a detailed
figure legend</p>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
